token/program-2022/src/extension.rs

```diff
@@ -38,30 +229,104 @@ pub enum Extension {
     /// Includes an optional mint close authority
     MintCloseAuthority,
 }
+impl TryFrom<u8> for ExtensionType {
+    type Error = ProgramError;
 
-/// Type-Length-Value Entry, used to encapsulate all extensions contained within an account
-#[repr(C)]
-#[derive(Clone, Copy, Debug, PartialEq)]
-pub struct TlvEntry<V> {
-    /// Extension type encoded in the rest of the entry
-    pub extension: Extension,
-    /// Length of the entry, in bytes
-    pub length: u32,
-    /// Deserialized value
-    pub value: V,
+    fn try_from(extension_type: u8) -> Result<Self, Self::Error> {
+        match extension_type {
```

Comment:
Have you tried https://crates.io/crates/enum-ordinalize before? I've used it before in programs to avoid exactly this kind of manual translation code

Reply:
I haven't, but it looks great, will do!

Reply:
Opting for `num-enum` since it's already being used in the program and works neatly